src/main/kotlin/no/nav/helse/dokument/api/dokumentV1Apis.kt

```diff
@@ -48,22 +47,26 @@ internal fun Route.dokumentV1Apis(
             throw Throwblem(ValidationProblemDetails(violations))
         }
 
-        val principal: JWTPrincipal = call.principal() ?: throw IllegalStateException("Principal ikke satt.")
+        val idToken = call.idToken()
 
         logger.trace("Dokument hentet fra reqeusten, forsøker å lagre")
-        val eier = eierResolver.hentEier(principal, dokument.eier!!.eiersFødselsnummer)
+        val eier = eierResolver.hentEier(idToken, dokument.eier!!.eiersFødselsnummer)
 
-        val dokumentId = when (val issuer = principal.payload.issuer) {
-            azureV1Issuer, azureV2Issuer -> dokumentService.lagreDokument(
+        val dokumentId = when {
+            idToken.issuerIsAzure() -> dokumentService.lagreDokument(
                 dokument = dokument.tilDokument(),
                 eier = eier,
                 medHold = true
             )
-            loginServiceV1Issuer, loginServiceV2Issuer -> dokumentService.lagreDokument(
+            idToken.issuerIsLoginservice() -> dokumentService.lagreDokument(
```

Comment:
Eventuelt legge til idToken som argument i when også sjekke direkte. Kan kjøre issuerIsTokendings() og issuerIsLoginservice() på samme linje ser det ut til

Reply:
Hvis man legger til idToken som argument på when, er det fordi man ønsker å sjekke typen på idToken i en av casene. E.g. is IdToken eller noe liknende. Kan ikke legge idToken som argument og kalle issuerIsLoginservice() uten å bruke idToken. issuerIsLoginservice(). Da blir det bare unødvendig. 

Men issuerIsLoginservice() og issuerIsTokendings() kan flyttes på samme linje ja :)

Reply:
Ah ja du har rett, jeg tenkte feil på hvordan argumentet ble brukt

Reply:
NP :)